.github/workflows/tests_linux.yml

```diff
@@ -14,124 +14,120 @@ env:
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v3
         with:
           files: ./BuildCov/coverage.info
           fail_ci_if_error: true
 
-  cpptestswithblas:
-    name: C++ tests (Linux, OpenBLAS)
-    runs-on: ${{ matrix.os }}
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: ubuntu-tests-reports
+          path: BuildCov/tests/results/report_cpptests.xml
+
+  pythontests:
     strategy:
       matrix:
         os: [ubuntu-20.04]
+
+    name: Python tests
+    runs-on: ${{ matrix.os }}
+
     steps:
-      - name: Cancel previous runs
-        uses: styfle/cancel-workflow-action@0.10.0
+      - name: Checkout PennyLane-Lightning
+        uses: actions/checkout@v2
         with:
-          access_token: ${{ github.token }}
+          path: main
+          fetch-depth: 2
 
       - uses: actions/setup-python@v2
         name: Install Python
         with:
           python-version: '3.7'
 
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 2
-
       - name: Install dependencies
-        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc-10 g++-10 libopenblas-dev gcovr lcov
+        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc-10 g++-10
 
-      - name: Build and run unit tests
+      - name: Get required Python packages
         run: |
-            cmake . -BBuild  -DCMAKE_BUILD_TYPE=Release -DENABLE_BLAS=ON -DBUILD_TESTS=1 -DCMAKE_CXX_COMPILER="$(which g++-$GCC_VERSION)"
-            cmake --build ./Build
-            mkdir -p ./Build/tests/results
-            ./Build/pennylane_lightning/src/tests/runner --order lex --reporter junit --out ./Build/tests/results/report.xml
-
-      - name: Upload test results
-        uses: actions/upload-artifact@v2
-        if: always()
-        with:
-          name: ubuntu-test-report
-          path: Build/tests/results/report.xml
-
-      - name: Build and run unit tests for code coverage
+          cd main
+          python -m pip install --upgrade pip
+          pip install -r requirements-dev.txt
+      - name: Install ML libraries for interfaces
         run: |
-            cmake pennylane_lightning/src -BBuildCov  -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON -DENABLE_COVERAGE=ON -DENABLE_BLAS=ON -DCMAKE_CXX_COMPILER="$(which g++-$GCC_VERSION)"
-            cmake --build ./BuildCov
-            cd ./BuildCov
-            ./tests/runner
-            lcov --directory . -b ../pennylane_lightning/src --capture --output-file coverage.info
-            lcov --remove coverage.info '/usr/*' --output-file coverage.info
-
+          pip install --upgrade torch==$TORCH_VERSION -f https://download.pytorch.org/whl/cpu/torch_stable.html
+          pip install --upgrade "jax[cpu]"  # This also installs jaxlib
+          pip install --upgrade tensorflow~=$TF_VERSION keras~=$TF_VERSION
```

Comment:
We should check which versions are tested by default in PennyLane now, as there was a recent update to the packages

Reply:
Makes sense, I will do that.